services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/validation/HankePublicValidatorTest.kt

```diff
@@ -63,29 +232,86 @@ internal class HankePublicValidatorTest {
     }
 
     @Test
-    fun `No errors when rakennuttajat missing`() {
+    fun `when rakennuttajat missing should return ok`() {
         val result =
-            HankePublicValidator.validateHankeHasMandatoryFields(
+            validateHankeHasMandatoryFields(
                 completeHanke().apply { rakennuttajat = mutableListOf() }
             )
 
         assertTrue(result.isOk())
     }
 
     @Test
-    fun `No errors when toteuttajat missing`() {
+    fun `when toteuttajat missing should return ok`() {
         val result =
-            HankePublicValidator.validateHankeHasMandatoryFields(
-                completeHanke().apply { toteuttajat = mutableListOf() }
-            )
+            validateHankeHasMandatoryFields(completeHanke().apply { toteuttajat = mutableListOf() })
+
+        assertTrue(result.isOk())
+    }
+
+    @Test
+    fun `when ytunnus is present and valid should return ok`() {
+        val hanke = completeHanke()
+
+        val result = validateHankeHasMandatoryFields(hanke)
+
+        val ytunnusCount = hanke.extractYhteystiedot().mapNotNull { it.ytunnus }.count()
+        assertThat(ytunnusCount).isGreaterThanOrEqualTo(1)
+        assertTrue(result.isOk())
+    }
+
+    @Test
+    fun `when ytunnus is present and not valid should return not ok`() {
```

Comment:
This might be something we want in HankeValidator. I don't think there's a reason we should allow anyone to save a non-valid y-tunnus.

Reply:
Added following logic to hankevalidator: 
- If ytunnus is not null, it must be valid
- If tyyppi is YKSITYISHENKILO, ytunnus must be null.